HW0/P4/P4.py

```diff
@@ -40,37 +57,79 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    m = np.loadtxt('P4_measurements.txt',delimiter=',')
+    s_obs = np.dot(m, np.diag([1.0/rx, 1.0/ry, 1.0/rz]))
+    x_coords = s_obs[:,0]
+    y_coords = s_obs[:,1]
+    z_coords = s_obs[:,2]
+
+    ax.plot(x_coords, y_coords, z_coords,
+            '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.diag(np.ones(6))
+    for ii in range(3):
+        A[ii, ii+3] = dt
+        A[ii+3, ii+3] -= c * dt
```

Comment:
It took me a little while to understand the pattern that you were using to fill out the matrix here, but this is a nice clean way to fill in A.